src/app/Login/page.tsx

```diff
@@ -2,40 +2,88 @@
 import React, { useState } from 'react';
 import ActiveButton from '@/components/Active-Button/ActiveButton';
 import InactiveButton from '@/components/Inactive-Button/InactiveButton';
-import InputField from '@/components/Input-textbox/textbox';
+import InputField from '@/components/textbox/textbox';
 import styles from './page.module.css';
 import { emailRegex } from '@/utils/constants';
 
 const LoginPage: React.FC = () => {
   // State for email and password
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [emailerror, EsetError] = useState(' ');
+  const [passerror, PsetError] = useState(' ');
+  const [shakeEmail, setShakeEmail] = useState(false);
+  const [shakePassword, setShakePassword] = useState(false);
 
-  // Handlers for input changes
   const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const value = e.target.value;
     setEmail(value);
+    if (!value) {
+      EsetError('Email cannot be empty.');
+    } else if (!emailRegex.test(value)) {
+      EsetError('Invalid email address.');
+    } else {
+      EsetError(''); // Clear error if valid
+    }
```

Comment:
Do not set this to trigger on change, errors should only submit on error once the user's input is final, not as the user is still trying to form their input. Prematurely showing an error is not good for the user experience